src/org/thoughtcrime/securesms/components/reminder/DozeReminder.java

```diff
@@ -101,4 +105,19 @@ public static void dozeReminderTapped(Context context) {
       context.startActivity(intent);
     }
   }
+
+  public static void maybeAskDirectly(Context context) {
+    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M
```

Comment:
i think, we should put the while function body to a try/catch block, eg. https://stackoverflow.com/search?page=3&tab=Relevance&q=ACTION_REQUEST_IGNORE_BATTERY_OPTIMIZATIONS reports several situations where the things do not work and i can easily imagine that things may crash - esp. as this area is highly modified by the manufacturers.

Reply:
i changed that.